x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -211,7 +214,13 @@ export function AddSettingsFlyout({
 }
 async function deleteConfig(selectedConfig: Config) {
   try {
-    await deleteAgentConfiguration(selectedConfig.id);
+    await callApmApi({
+      pathname: '/api/apm/settings/agent-configuration/{configurationId}',
+      method: 'DELETE',
+      params: {
+        path: { configurationId: selectedConfig.id }
+      }
+    });
```

Comment:
Hmm... I think it is a code smell when the component has to know details about the transport. I think encapsulating that in a method like `eleteAgentConfiguration(selectedConfig.id)` reads a lot better.

Reply:
I'm open to having the data-fetching hooks use `callApmApi` directly but I really think it's a bad idea to put this into components.

Reply:
The function is already called `deleteConfig` though (just like there's a `saveConfig` function). What do you think the abstraction `deleteAgentConfiguration` offers on top of that? You'd rather have the API call in a separate file?

Reply:
I remember the hooks => rest helper methods => callApi were confusing for me to process. Agree that the `callApmApi` calls are not easy on the eye. Can we do it with just hooks? Or maybe just helper methods?